models/DataObject/Localizedfield/Dao.php

```diff
@@ -646,32 +646,45 @@ public function createLocalizedViews()
         /**
          * macro for creating ifnull statement
          *
-         * @param string $field
+         * @param string $row
          * @param array $languages
          *
          * @return string
          */
-        $getFallbackValue = function ($field, array $languages) use (&$getFallbackValue, $db) {
+        $getFallbackValue = function ($row, array $languages) use (&$getFallbackValue, $db) {
 
             // init
             $lang = array_shift($languages);
-
+            $field = $row['Field'];
             // get fallback for current language
             $fallback = count($languages) > 0
-                ? $getFallbackValue($field, $languages)
+                ? $getFallbackValue($row, $languages)
                 : 'null';
 
             // create query
-            $sql = sprintf(
-                'IF(`%s`.`%s` IS NULL OR `%s`.`%s` = "", %s, `%s`.`%s`)',
-                $lang,
-                $field,
-                $lang,
-                $field,
-                $fallback,
-                $lang,
-                $field
-            );
+            if ($row['Type'] === 'tinyint(1)' || $row['Type'] === 'int(4}') {
```

Comment:
```suggestion
            if ($row['Type'] === 'tinyint(1)' || $row['Type'] === 'int(4)') {
```
But probably it's better anyway to not include the length. @weisswurstkanone wdyt?